### ISSUE TITLE
I am utterly confused by the compose instructions.

### ISSUE THREAD

**Issue by scyto** [cuigh/swirl#40]
I am a little confused by the instructions and have been unable to get this working.  I am unclear if i have hit an issue, didn't read something i should have (like manually creating databases and users)?

I am trying to deploy in a swarm using mongodb, i am unclear if mongodb is in the docker image or not?
I have tried both of the following:

- assuming mongodb is in the image
- assuming i need separate mongodb

(note i have never used mongodb before)
for #1 aobove

```
panic: failed to load storage engine: no reachable servers
```
as such i believe there to be no server running in the container

for #2 above this is mycompose below, when swirl tries to connect to the mongodb it results in this if the db ends in /swirl

```
panic: failed to load storage engine: server returned error on SASL authentication step: Authentication failed.
```

if it doesn't end in swirl i get 

```
panic: failed to load storage engine: server returned error on SASL authentication step: BSON field 'saslContinue.mechanism' is an unknown field.
```

version: '3.8'

services:
  swirl:
    image: cuigh/swirl
    environment:
      DB_TYPE: mongo
      DB_ADDRESS: mongodb://mongoroot:password@mongodb:27017/swirl
      DOCKER_ENDPOINT: tcp://swirl_manager_agent:2375
      AGENTS: swirl_manager_agent,swirl_worker_agent
    ports:
      - 8001:8001
    deploy:
      replicas: 1
      placement:
        constraints: [ node.role == worker ]

  manager_agent:
    image: cuigh/socat
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    deploy:
      mode: global
      placement:
        constraints: [ node.role == manager ]

  worker_agent:
    image: cuigh/socat

    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    deploy:
      mode: global
      placement:
        constraints: [ node.role == worker ]
        
  mongodb:
    image: mongo
    restart: always
    environment:
      MONGO_INITDB_ROOT_USERNAME: mongoroot
      MONGO_INITDB_ROOT_PASSWORD: password

  mongo-express:
    image: mongo-express
    restart: always
    ports:
      - 7081:8081
    environment:
      ME_CONFIG_MONGODB_ADMINUSERNAME: mongoroot
      ME_CONFIG_MONGODB_ADMINPASSWORD: password
      ME_CONFIG_MONGODB_URL: mongodb://mongoroot:password@mongodb:27017/      
```

**Comment by cuigh** [comment]
For approach 1, DB_ADDRESS should be: mongodb://mongoroot:password@**swirl_mongodb**:27017/swirl. (Assuming compose name is swirl)

**Comment by scyto** [comment]
Thanks for the reply.

Do you mean approach two?  There is only a service called mongodb in approach two - your reference docker compose has no mongodb service. I will take this as confirmation that you docs assume a pre-setup mongodb - might be good to actually call that out.

You seem to be unaware that in a swarm stack you can call the service just by the service name  - I assure you one can `ping mongodb` from any container node in the stack just fine - it doesn't need the stack_ prefix :-)

It still appears to me swirl doesnt support the latest auth mechanisms correctly in mongo - but i am making a lot of un-educated guesses to get to that conclusion!  Again maybe its because i didn't do something i should on mongodb?

I also now realize with a prometheus setup this is an incomplete solution.

tl;dr This modified version below got me up and running to see what swirl is all about with the downside there is no no auth support.

```
version: '3.8'

services:
  swirl:
    image: cuigh/swirl
    environment:
      DB_TYPE: mongo
      DB_ADDRESS: mongodb://mongodb:27017/swirl    
      DOCKER_ENDPOINT: tcp://swirl_manager_agent:2375
      AGENTS: swirl_manager_agent,swirl_worker_agent
    ports:
      - 8001:8001
    deploy:
      replicas: 1
      placement:
        constraints: [ node.role == worker ]

  manager_agent:
    image: cuigh/socat
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    deploy:
      mode: global
      placement:
        constraints: [ node.role == manager ]

  worker_agent:
    image: cuigh/socat

    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    deploy:
      mode: global
      placement:
        constraints: [ node.role == worker ]
        
  mongodb:
    image: mongo
    restart: always
#    environment:
#      MONGO_INITDB_ROOT_USERNAME: mongoroot
#      MONGO_INITDB_ROOT_PASSWORD: password

  mongo-express:
    image: mongo-express
    restart: always
    ports:
      - 7081:8081
    environment:
#      ME_CONFIG_MONGODB_ADMINUSERNAME: mongoroot
#      ME_CONFIG_MONGODB_ADMINPASSWORD: password
      ME_CONFIG_MONGODB_URL: mongodb://mongodb:27017/   
```

**Comment by cuigh** [comment]
Thank you for your advice. You are right, Swirl's document should really have a pre requirements chapter. I will add this part later and provide a composite file out of the box.
